samples/discoverNatBehavior.c

```diff
@@ -2,108 +2,115 @@
 
 #define NETWORK_INTERFACE_NAME_PARAM "-i"
 #define STUN_HOSTNAME_PARAM          "-s"
-#define DEFAULT_STUN_HOST            "stun:stun.sipgate.net:3478"
 
 BOOL filterFunc(UINT64 data, PCHAR name)
 {
     CHAR* desiredInterface = (CHAR*) data;
-    if (desiredInterface == NULL || STRCMP(name, desiredInterface) == 0) {
+    if (desiredInterface == NULL || STRNCMP(name, desiredInterface, STRLEN(name)) == 0) {
         return TRUE;
     }
     return FALSE;
 }
 
 INT32 main(INT32 argc, CHAR** argv)
 {
-    PCHAR interfaceName = NULL, stunHostname = NULL;
-    printf("Usage: ./discoverNatBehavior -i network-interface-name -s stun-hostname\n");
+    PCHAR interfaceName = NULL, stunHostname = NULL, pRegion = NULL, pHostnamePostfix = NULL;
     NAT_BEHAVIOR mappingBehavior = NAT_BEHAVIOR_NONE, filteringBehavior = NAT_BEHAVIOR_NONE;
     INT32 i;
     UINT32 logLevel = LOG_LEVEL_DEBUG;
     STATUS status = STATUS_SUCCESS;
     CHAR* logLevelStr = NULL;
 
+    if ((logLevelStr = GETENV(DEBUG_LOG_LEVEL_ENV_VAR)) != NULL) {
+        status = STRTOUI32(logLevelStr, NULL, 10, &logLevel);
+        if (STATUS_FAILED(status)) {
+            DLOGE("Failed to parse log level with error 0x%08x", status);
+            exit(1);
+        }
+        logLevel = MIN(MAX(logLevel, LOG_LEVEL_VERBOSE), LOG_LEVEL_SILENT);
+    }
+    SET_LOGGER_LOG_LEVEL(logLevel);
+
+    DLOGI("Usage: ./discoverNatBehavior -i network-interface-name -s stun-hostname");
+
+    if ((pRegion = GETENV(DEFAULT_REGION_ENV_VAR)) == NULL) {
+        pRegion = DEFAULT_AWS_REGION;
+    }
+    pHostnamePostfix = KINESIS_VIDEO_STUN_URL_POSTFIX;
+    // If region is in CN, add CN region uri postfix
+    if (STRSTR(pRegion, "cn-")) {
+        pHostnamePostfix = KINESIS_VIDEO_STUN_URL_POSTFIX_CN;
+    }
+    stunHostname = (PCHAR) MEMALLOC((MAX_ICE_CONFIG_URI_LEN + 1) * SIZEOF(CHAR));
+    SNPRINTF(stunHostname, MAX_ICE_CONFIG_URI_LEN + 1, KINESIS_VIDEO_STUN_URL, pRegion, pHostnamePostfix);
+
     for (i = 1; i < argc; ++i) {
-        if (STRCMP(argv[i], NETWORK_INTERFACE_NAME_PARAM) == 0) {
+        if (STRNCMP(argv[i], NETWORK_INTERFACE_NAME_PARAM, STRLEN(argv[i])) == 0) {
```

Comment:
same here use `STRNLEN` especially here since it's user supplied input.

Reply:
Done